src/Promise.fs

```diff
@@ -454,6 +454,142 @@ let Parallel (pr: seq<JS.Promise<'T>>): JS.Promise<'T[]> = jsNative
 /// <returns>Return a new promise returning an array containing all the promise result</returns>
 let all (pr: seq<JS.Promise<'T>>): JS.Promise<'T[]> = jsNative
 
+[<StringEnum>]
+type SettledStatus =
+    | Fulfilled
+    | Rejected
+
+type SettledValue<'T> = 
+    {| status: SettledStatus; value: 'T option; reason: exn option |}
```

Comment:
I had some issues when using anonymous records in bindings, because the F# complains sometimes when they're in different assemblies and used in certain ways ([example](https://github.com/dotnet/fsharp/issues/7362)). So I tend to avoid them now, maybe just an interface instead?

```fsharp
type SettledValue<'T> =
    status: SettledStatus
    value: 'T option
    reason: exn option
```

Reply:
Ah yeah, you're definitely right I'll change that